src/Emanote/Pandoc/BuiltinFilters.hs

```diff
@@ -51,3 +54,35 @@ fixEmojiFontFamily =
             newAttrs = attrs <> one emojiFontAttr
          in B.Span (id', classes, newAttrs) is
     x -> x
+
+-- Adds a data-linkicon=external attribute to external links that contain some text in their description, provided that they do not already have a data-linkicon attribute.
+setExternalLinkicon :: W.Walkable B.Inline b => b -> b
+setExternalLinkicon =
+  W.walk $ \case
+    B.Link (id', classes, attrs) inlines (url, title)
+      | hasURIScheme url && containsText inlines ->
+        let showLinkIconAttr = ("data-linkicon", "external")
+            newAttrs = insert attrs showLinkIconAttr
+         in B.Link (id', classes, newAttrs) inlines (url, title)
+    x -> x
+  where
+    insert as a
+      | fst a `elem` (fst <$> as) = as
+      | otherwise = a : as
+    hasURIScheme =
+      isRight . P.parse schemeP ""
+      where
+        schemeP = do
+          c <- PC.letter
+          cs <- P.many (PC.alphaNum P.<|> P.oneOf ".-+")
+          _ <- PC.char ':'
+          return (c : cs)
+    containsText =
```

Comment:
Please a type-signature and, more importantly, comments as to what this function does.

Reply:
Fixed.